### PR TITLE
Add IS_FILE_RENDERED filter to enhance file processing flexibility

### PR DESCRIPTION
**Pull request by Abdul-Muqadim-Arbisoft**
**Description:**
This pull request introduces a new filter, IS_FILE_RENDERED, designed to enhance the flexibility of file handling within Tutor plugins. The new IS_FILE_RENDERED filter allows plugin developers to specify files that should not be rendered but instead copied .

**Implementation Details:**
The IS_FILE_RENDERED filter works by accepting a boolean flag and a file path. It returns a boolean indicating whether the file should be rendered. By default, all files are set to be rendered unless explicitly specified .

### CODE REVIEW

File: tutor/env.py [hunk 1]
```diff
@@ -551,3 +550,11 @@ def _delete_plugin_templates(plugin: str, root: str, _config: Config) -> None:
                 raise exceptions.TutorError(
                     f"Could not delete file {e.filename} from plugin {plugin} in folder {path}"
                 )
+
+
+@hooks.Filters.IS_FILE_RENDERED.add()
+def _do_not_render_binary_files(result: bool, path: str) -> bool:
+    """Return bianry file"""
```
Comment (DawoudSheraz) [inline tutor/env.py:557]: nit: the docstring does not make sense. Please update it.

File: changelog.d/20240508_111948_abdul.muqadim_is_file_rendered_filter.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- [Feature] Introduces the IS_FILE_RENDERED Filter, enabling developers to specify files that should be copied without rendering. (by @Abdul-Muqadim-Arbisoft)
```
Comment (DawoudSheraz) [inline changelog.d/20240508_111948_abdul.muqadim_is_file_rendered_filter.md:1]: nit: having a bit more context here can be helpful for the devs.

**Comment by hinakhadim** [comment]
@DawoudSheraz @regisb  I have tested this and working fine. Can you plz review it?

File: changelog.d/20240508_111948_abdul.muqadim_is_file_rendered_filter.md (hunk 2 above)
Comment (Abdul-Muqadim-Arbisoft) [reply changelog.d/20240508_111948_abdul.muqadim_is_file_rendered_filter.md]: concern addressed

File: tutor/env.py (hunk 1 above)
Comment (Abdul-Muqadim-Arbisoft) [reply tutor/env.py]: concern addressed

**Review by DawoudSheraz — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
@hinakhadim This is waiting on your review, thanks.

**Review by hinakhadim — APPROVED** [review]

**Pull request merged**